bootstrap/kubeadm/types/upstreamv1beta2/conversion.go

```diff
@@ -92,3 +92,9 @@ func Convert_v1beta1_JoinConfiguration_To_upstreamv1beta2_JoinConfiguration(in *
 	// JoinConfiguration.Patches does not exist in kubeadm v1beta2 API
 	return autoConvert_v1beta1_JoinConfiguration_To_upstreamv1beta2_JoinConfiguration(in, out, s)
 }
+
```

Comment:
Okay let's start with some context.

We have conversion fuzz tests which are testing if we lose data during conversion.

Let's look at a specific one: `sigs.k8s.io/cluster-api/bootstrap/kubeadm/types/upstreamv1beta1: TestFuzzyConversion/for_InitConfiguration`

This test tests:
* hub-spoke-hub
  * Create a bootstrapv1.InitConfiguration
  * Fuzz it, aka fill in some random values for fields
  * Convert it to upstreamv1beta1.InitConfiguration  (the type used by kubeadm)
  * Convert the result back to bootstrapv1.InitConfiguration
  * Check if we lost data
* spoke-hub-spoke
  * Create a upstreamv1beta1.InitConfiguration
  * Fuzz it, aka fill in some random values for fields
  * Convert it to bootstrapv1.InitConfiguration  (the type used by the kubeadm bootstrap provider)
  * Convert the result back to upstreamv1beta1.InitConfiguration
  * Check if we lost data

In those unit tests we don't care about loosing data, because the only thing the bootstrap provider will ever do is take the `bootstrapv1.InitConfiguration` and convert it to `upstreamv1beta1.InitConfiguration` (same for similar types). We never convert back.

The tests are now failing because they are testing full roundtrips (what I described above). Because ImagePullPolicy doesn't exist in in the `upstreamv1beta1` types, we lose date in the `hub-spoke-hub` case.

Reply:
What we now do to fix this test is to ensure that when we fill in the initial data in the `bootstrapv1.InitConfiguration` type, ImagePullPolicy is always empty, so there is no data to lose.

Concrete changes:

* bootstrap/kubeadm/types/upstreamv1beta1/conversion_test.go
  * Add the following to `kubeadmNodeRegistrationOptionsFuzzer`:
    ```go
	// NodeRegistrationOptions.ImagePullPolicy does not exist in kubeadm v1beta1 API, so setting it to nil in order to avoid round trip errors.
	obj.ImagePullPolicy = ""
    ```
* bootstrap/kubeadm/types/upstreamv1beta2/conversion_test.go
  *  Add:
  ```go
  func kubeadmNodeRegistrationOptionsFuzzer(obj *bootstrapv1.NodeRegistrationOptions, c fuzz.Continue) {
  	c.FuzzNoCustom(obj)

	// NodeRegistrationOptions.ImagePullPolicy does not exist in kubeadm v1beta2 API, so setting it to nil in order to avoid round trip errors.
	obj.ImagePullPolicy = ""
  }
  ```
  * Also add it to the list in `fuzzFuncs`

Reply:
Now that I thought a bit about what we're doing I'll see if we can simplify the whole thing in a follow-up PR